functions.sh

```diff
@@ -61,4 +61,146 @@ function saveTaskStatus() {
   fi
 }
 
+function getDecryptedCredential() {
+    local fernet_key="$1"
+    local encrypted_value="$2"
+
+    python3 -c "
+from cryptography.fernet import Fernet
+
+fernet_key = '$fernet_key'
+f = Fernet(fernet_key.encode('utf-8'))
+encrypted_value = b'$encrypted_value'
+
+try:
+    decrypted_value = f.decrypt(encrypted_value)
+    print(decrypted_value.decode())
+except Exception as e:
+    print(f'Decryption error: {e}')
+"
+}
+
+function passwordStrengthChecker() {
+    local password="$1"
+
+    if [ -z "$password" ]; then
+      logErrorMessage "password cannot be empty."
+      exit 1
+    fi
+
+    if [ "${#password}" -ge 8 ] &&  [[ $password == *[A-Za-z]* ]] &&  [[ $password == *[0-9]* ]] && [[ $password == *['#?!@$\ %^\&*-']* ]]
+    then
+      true
+    else
+      logErrorMessage "Weak password. Please ensure the password meets the criteria: at least 8 characters, one uppercase letter, one lowercase letter, one digit, and one special character."
+      exit 1
+    fi
+}
+
+function validDBname() {
+    local db_name="$1"
+    
+    if [ -z "$db_name" ]; then
+        logErrorMessage "Database name cannot be empty."
+        exit 1
+    fi
+    
+    if [[ "$db_name" =~ ^[a-zA-Z0-9_]+$ ]]; then
+        true
+    else
+        logErrorMessage "Invalid characters in the database name [$db_name]. Only alphanumeric characters (a-zA-Z), digits (1), underscore (_) are allowed."
+        exit 1
+    fi
+}
+
+function validDBusername() {
+    local username="$1"
+    
+    if [ -z "$username" ]; then
+        logErrorMessage "Username name cannot be empty."
+        exit 1
+    fi
+    
+    if [[ "$username" =~ ^[a-zA-Z0-9_]+$ ]]; then
+        true
+    else
+        logErrorMessage "Invalid characters in the username [$username]. Only alphanumeric characters (a-zA-Z), digits (1), underscore (_) are allowed."
+        exit 1
+    fi
+}
+
+function getEncryptedCredential() {
+    local credentialManagement="$1"
+    local credentialKey="$2"
+    
+    encrypted_value=$(echo "$credentialManagement" | jq -r ".$credentialKey")
+    echo "$encrypted_value"
+}
+
+function mysqlFilterWarning() {
+    sed '/mysql: \[Warning\] Using a password on the command line interface can be insecure\./d'
+}
+
+function mysqlcheckDatabaseConnection() {
```

Comment:
Let's make separate library like `database.sh` and add generic functions like `testConnection`, `getUser` etc. This will serve as single library to connect with multiple db types.